docs/design.md

```diff
@@ -1,37 +1,145 @@
+    exporters: [opencensus, jaeger, stackdriver, zipkin]
+```
+
+The above example defines a pipeline for “traces” type of telemetry data, with 3 receivers, 4 processors and 4 exporters. 
+
+For details of config file format see [this document](https://docs.google.com/document/d/1GWOzV0H0RTN1adiwo7fTmkjfCATDDFGuOB4jp3ldCc8/edit#).
+
+### Receivers
+
+Receivers typically listen on a network port and receive telemetry data. Usually one receiver is configured to send received data to one pipeline, however it is also possible to configure the same receiver to send the same received data to multiple pipelines. This can be done by simply listing the same receiver in the “receivers” key of several pipelines:
+
+```yaml
+receivers:
+  opencensus:
+    endpoint: "127.0.0.1:55678”
+
+pipelines:
+  traces:  # a pipeline of “traces” type
+    receivers: [opencensus]
+    processors: [tags, tail-sampling, batch, queued-retry]
+    exporters: [jaeger]
+  traces/2:  # another pipeline of “traces” type
+    receivers: [opencensus]
+    processors: [batch]
+    exporters: [opencensus]
+```
+
+In the above example “opencensus” receiver will send the same data to pipeline “traces” and to pipeline “traces/2”. (Note: the configuration uses composite key names in the form of `type[/name]` as defined in this [this document](https://docs.google.com/document/d/1GWOzV0H0RTN1adiwo7fTmkjfCATDDFGuOB4jp3ldCc8/edit#)).
+
+When the Service loads this config the result will look like this (part of processors and exporters are omitted from the diagram for brevity):
+
+![Receivers](images/design-receivers.png)
+
+Important: when the same receiver is referenced in more than one pipeline the Service will create only one receiver instance at runtime that will send the data to fanoutprocessor which in turn will send the data to the first processor of each pipeline. The data propagation from receiver to fanoutprocessor and then to processors is via synchronous function call. This means that if one processor blocks the call the other pipelines that are attached to this receiver will be blocked from receiving the same data and the receiver itself will stop processing and forwarding newly received data.
+
+### Exporters
+
+Exporters typically forward the data they get to a destination on a network (but they can also send it elsewhere, e.g “logging” exporter writes the telemetry data to a local file). 
+
+The configuration allows to have multiple exporters of the same type, even in the same pipeline. For example one can have 2 “jaeger” exporters defined each one sending to a different jaeger endpoint, e.g.:
+
+```yaml
+exporters:
+  jaeger/1:
+    protocols:
+      grpc:
+        endpoint: "example.com:14250”
+  jaeger/2:
+    protocols:
+      grpc:
+        endpoint: "127.0.0.1:14250”
+```
+
+Usually an exporter gets the data from one pipeline, however it is possible to configure multiple pipelines to send data to the same exporter, e.g.:
+
+```yaml
+exporters:
+  jaeger:
```

Comment:
I think the config for Jaeger exporter has been updated to:
```yaml
exporters:
  jaeger-thrift-http:
  jaeger-grpc:
```
E.g https://github.com/open-telemetry/opentelemetry-service/blob/master/exporter/jaeger/jaegergrpcexporter/testdata/config.yaml.

Reply:
Thanks for catching, I will fix.

Reply:
Fixed.